### PR TITLE
Add support for CSI snapshot beta APIs

### PR DESCRIPTION
**Pull request by PrasadG193**
## Change Overview

Add support for CSI snapshot beta APIs
Refactor snapshot tests to support beta snapshots

## Pull request type

Please check the type of change your PR introduces:
- [x] :rainbow: Refactoring (no functional changes, no api changes)
- [x] :sunflower: Feature

## Issues

- #XXX

## Test Plan

Steps to test on minikube K8s 1.17:
```
Step 1: Deploy k8s 1.17 with minikube
$ minikube start --kubernetes-version='v1.17.0'

Step 2: Install csi hostpath driver
Follow this guide: https://github.com/kubernetes-csi/csi-driver-host-path/blob/master/docs/deploy-1.17-and-later.md

Step 3: Create StorageClass  and VolumeSnapshotClass
$ kubectl create -f https://raw.githubusercontent.com/kubernetes-csi/csi-driver-host-path/master/examples/csi-storageclass.yaml
$ kubectl create -f https://raw.githubusercontent.com/kubernetes-csi/csi-driver-host-path/master/deploy/kubernetes-1.17/snapshotter/csi-hostpath-snapshotclass.yaml

Step 4: Run snapshot tests
$ cd pkg/kube/snapshot/
$ go test -v -check.vv .
```

Test logs:
```
$ go test -v -check.vv .
=== RUN   Test
START: snapshot_test.go:75: SnapshotTestSuite.SetUpSuite
PASS: snapshot_test.go:75: SnapshotTestSuite.SetUpSuite 3.437s

START: snapshot_test.go:429: SnapshotTestSuite.TestGetVolumeSnapshotClassFake
PASS: snapshot_test.go:429: SnapshotTestSuite.TestGetVolumeSnapshotClassFake    0.000s

START: snapshot_test.go:379: SnapshotTestSuite.TestNewSnapshotter
PASS: snapshot_test.go:379: SnapshotTestSuite.TestNewSnapshotter        0.000s

START: snapshot_test.go:242: SnapshotTestSuite.TestVolumeSnapshotAlpha
SKIP: snapshot_test.go:242: SnapshotTestSuite.TestVolumeSnapshotAlpha (No Storageclass with CSI provisioner, install CSI and create a storageclass for it)

START: snapshot_test.go:253: SnapshotTestSuite.TestVolumeSnapshotBeta
VolumeSnapshot test - source namespace: snapshot-test-source-66327 - target namespace: snapshot-test-target-66327
PASS: snapshot_test.go:253: SnapshotTestSuite.TestVolumeSnapshotBeta    303.302s

START: snapshot_test.go:161: SnapshotTestSuite.TestVolumeSnapshotCloneFake
PASS: snapshot_test.go:161: SnapshotTestSuite.TestVolumeSnapshotCloneFake       0.005s

START: snapshot_test.go:119: SnapshotTestSuite.TestVolumeSnapshotFake
PASS: snapshot_test.go:119: SnapshotTestSuite.TestVolumeSnapshotFake    0.002s

START: snapshot_test.go:114: SnapshotTestSuite.TearDownSuite
Failed to list snapshots, Namespace: snapshot-test-source-66327, Error: the server could not find the requested resource
Skipping deleting the namespace, Namespace: snapshot-test-source-66327
Failed to list snapshots, Namespace: snapshot-test-target-66327, Error: the server could not find the requested resource
Skipping deleting the namespace, Namespace: snapshot-test-target-66327
PASS: snapshot_test.go:114: SnapshotTestSuite.TearDownSuite     3.094s

OK: 5 passed, 1 skipped
--- PASS: Test (309.84s)
PASS
ok      github.com/kanisterio/kanister/pkg/kube/snapshot        309.866s
```

- [x] :muscle: Manual
- [x] :zap: Unit test
- [ ] :green_heart: E2E

### CODE REVIEW

File: pkg/kube/snapshot/snapshot_alpha.go [hunk 1]
```diff
@@ -290,19 +231,66 @@ func (sna *SnapshotAlpha) createVolumeSnapshot(name, namespace string, pvcObject
 				"name":      name,
 				"namespace": namespace,
 			},
+		},
+	}
+	if pvcName != "" {
+		snap.Object["spec"] = map[string]interface{}{
+			"source": map[string]interface{}{
+				"kind":      pvcKind,
```
Comment (bathina2) [inline pkg/kube/snapshot/snapshot_alpha.go]: I feel like you should capitalize "PVCKind".

**Comment by bathina2** [comment]
Otherwise looks good to me, however you it's quite a big change you should get a second opinion from Tom.

File: pkg/kube/snapshot/snapshot_alpha.go [hunk 2]
```diff
@@ -317,3 +305,20 @@ func TransformUnstructured(u *unstructured.Unstructured, value interface{}) erro
 	}
 	return nil
 }
+
+func getSnapshotClassbyAnnotation(dynCli dynamic.Interface, gvr schema.GroupVersionResource, annotationKey, annotationValue string) (string, error) {
+	us, err := dynCli.Resource(gvr).Namespace("").List(metav1.ListOptions{})
```
[the remark this answers is not in the document]
Comment (PrasadG193) [reply pkg/kube/snapshot/snapshot_alpha.go]: No. My bad. Don't need to call `Namespace()` here.

File: pkg/kube/snapshot/snapshot_beta.go [hunk 3]
```diff
@@ -0,0 +1,326 @@
...
+	}
+
+	vs := &v1beta1.VolumeSnapshot{}
+	err = TransformUnstructured(us, vs)
+	if err != nil {
+		return nil, err
+	}
+	// Populate v1alpha1.VolumeSnapshot object from v1beta1.VolumeSnapshot
+	vsa := v1alpha1.VolumeSnapshot{}
+	vsa.ObjectMeta = vs.ObjectMeta
```
[the remark this answers is not in the document]
Comment (PrasadG193) [reply pkg/kube/snapshot/snapshot_beta.go]: Agree. Safer to use DeepCopy()

**Pull request merged**